src/helpers/delete-stale-branches.ts

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { HelperInputs } from '../types/generated';
+import { context } from '@actions/github';
+import * as core from '@actions/core';
+import { octokit } from '../octokit';
+import { map } from 'bluebird';
+import { paginateAllOpenPullRequests } from '../utils/paginate-open-pull-requests';
+import { PullRequestBranchesList } from '../types/github';
+
+export class DeleteStaleBranches extends HelperInputs {
+  days?: string;
+}
+
+export const deleteStaleBranches = async ({ days = '30' }: DeleteStaleBranches = {}) => {
+  const openPullRequests = await paginateAllOpenPullRequests();
+  const openPullRequestBranches = openPullRequests.map(pr => pr.head.ref);
+  const allBranches = await paginateAllUnprotectedBranches();
+  const branchesWithNoOpenPullRequest = allBranches.filter(({ name }) => !openPullRequestBranches.includes(name));
```

Comment:
mega mega nitpick but reducing runtime complexity on a somewhat heavy computation for big repos
```suggestion
  const openPullRequestBranches = new Set(openPullRequests.map(pr => pr.head.ref));
  const allBranches = await paginateAllUnprotectedBranches();
  const branchesWithNoOpenPullRequest = allBranches.filter(({ name }) => !openPullRequestBranches.has(name));
```